### PR TITLE
Introduce UseAnyOrNoneInsteadOfFind rule

### LINKED ISSUE #4243: Rule: Use `any` instead of `find`
## Expected Behavior of the rule
Use `any { /* predicate */ }` instead of `find { /* predicate */ } != null`

sample:

```kotlin
// non-compilant
listOf("static", "code").find { it.lenght == 4 } != null
// compilatn
listOf("static", "code").any { it.lenght == 4 }
```

## Context
I was making this "mistake" on #4241 and it should be easily sppoted by detekt.

For sure this is a style rule.

### PR DESCRIPTION
**Pull request by t-kameyama**
Fixes #4243

### CODE REVIEW

File: detekt-rules-style/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/UseAnyOrNoneInsteadOfFindSpec.kt [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+            val actual = subject.compileAndLintWithContext(env, code)
+            Assertions.assertThat(actual).hasSize(1)
+        }
+
+        it("Does not reports collections.lastOrNull != null") {
+            val code = "val x = listOf(1, 2, 3).lastOrNull { it == 4 } != null"
+            val actual = subject.compileAndLintWithContext(env, code)
+            Assertions.assertThat(actual).hasSize(0)
+        }
+    }
```
Comment (BraisGabin) [inline detekt-rules-style/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/UseAnyOrNoneInsteadOfFindSpec.kt:76]: Maybe add a test to check that we don't just flag all the `find` usages. Something like:

```kotlin
        it("Does not report collections.find") {
            val code = "val x = listOf(1, 2, 3).find { it == 4 }"
            val actual = subject.compileAndLintWithContext(env, code)
            Assertions.assertThat(actual).isEmpty()
        }
```

**Comment by BraisGabin** [comment]
As always, thanks for the contributions :)

File: detekt-rules-style/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/UseAnyOrNoneInsteadOfFindSpec.kt [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        }
+
+        it("Reports collections.find != null in extension") {
+            val code = "fun List<Int>.test(): Boolean = find { it == 4 } != null"
+            val actual = subject.compileAndLintWithContext(env, code)
+            Assertions.assertThat(actual).hasSize(1)
+        }
+
+        it("Does not reports collections.lastOrNull != null") {
+            val code = "val x = listOf(1, 2, 3).lastOrNull { it == 4 } != null"
```
[the remark this answers is not in the document]
Comment (t-kameyama) [reply detekt-rules-style/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/UseAnyOrNoneInsteadOfFindSpec.kt:72]: 382567d5c

File: detekt-rules-style/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/UseAnyOrNoneInsteadOfFindSpec.kt (hunk 1 above)
Comment (t-kameyama) [reply detekt-rules-style/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/UseAnyOrNoneInsteadOfFindSpec.kt:83]: df570ae1e

**Review by BraisGabin — APPROVED** [review]

File: detekt-rules-style/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/UseAnyOrNoneInsteadOfFindSpec.kt [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    setupKotlinEnvironment()
+
+    val env: KotlinCoreEnvironment by memoized()
+    val subject by memoized { UseAnyOrNoneInsteadOfFind() }
+
+    describe("UseAnyOrNoneInsteadOfFind rule") {
+        it("Reports collections.find != null") {
+            val code = "val x = listOf(1, 2, 3).find { it == 4 } != null"
+            val actual = subject.compileAndLintWithContext(env, code)
+            Assertions.assertThat(actual).hasSize(1)
```
Comment (cortinico) [inline detekt-rules-style/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/UseAnyOrNoneInsteadOfFindSpec.kt:20]: Can you import `org.assertj.core.api.Assertions.assertThat` so you won't have to repeat it in all the test?

**Review by cortinico — APPROVED** [review]

**Pull request merged**